packages/angular-material/test/table-control.spec.ts

```diff
@@ -0,0 +1,295 @@
+
+const uischema1: ControlElement = { type: 'Control', scope: '#' };
+const uischema2: ControlElement = {
+  type: 'Control',
+  scope: '#/properties/my'
+};
+const schema_object1 = {
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      foo: { type: 'string' },
+      bar: { type: 'string' }
+    }
+  }
+};
+const schema_object2 = {
+  type: 'object',
+  properties: {
+    my: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          foo: { type: 'string' },
+          bar: { type: 'string' }
+        }
+      }
+    }
+  }
+};
+const schema_simple1 = {
+  type: 'array',
+  items: {
+    type: 'string'
+  }
+};
+const schema_simple2 = {
+  type: 'object',
+  properties: {
+    my: {
+      type: 'array',
+      items: {
+        type: 'string'
+      }
+    }
+  }
+};
+const renderers = [
+  { tester: TextControlRendererTester, renderer: TextControlRenderer },
+  { tester: TableRendererTester, renderer: TableRenderer }
+];
+
+describe('Table tester', () => {
+  it('should succeed', () => {
+    expect(TableRendererTester(uischema1, schema_object1)).toBe(3);
+    expect(TableRendererTester(uischema1, schema_simple1)).toBe(3);
+    expect(TableRendererTester(uischema2, schema_object2)).toBe(3);
+    expect(TableRendererTester(uischema2, schema_simple2)).toBe(3);
+  });
+});
+describe('Table', () => {
+  let fixture: ComponentFixture<any>;
+  let component: any;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [TableRenderer, TextControlRenderer],
+      imports: [
+        CommonModule,
+        JsonFormsModule,
+        NgReduxModule,
+        MatCardModule,
+        NoopAnimationsModule,
+        MatFormFieldModule,
+        MatInputModule,
+        ReactiveFormsModule,
+        FlexLayoutModule,
+        MatTableModule
+      ],
+      providers: [{ provide: NgRedux, useFactory: MockNgRedux.getInstance }]
+    })
+      .overrideModule(BrowserDynamicTestingModule, {
+        set: {
+          entryComponents: [TextControlRenderer]
+        }
+      })
+      .compileComponents();
+
+    MockNgRedux.reset();
+    fixture = TestBed.createComponent(TableRenderer);
+    component = fixture.componentInstance;
+  }));
+
+  it('renders object array on root', async(() => {
+    const mockSubStore = MockNgRedux.getSelectorStub();
+    component.uischema = uischema1;
+    component.schema = schema_object1;
+
+    mockSubStore.next({
+      jsonforms: {
+        renderers: renderers,
+        core: {
+          data: [
+            { foo: 'foo_1', bar: 'bar_1' },
+            { foo: 'foo_2', bar: 'bar_2' }
+          ],
+          schema: schema_object1
+        }
+      }
+    });
+    mockSubStore.complete();
+    fixture.detectChanges();
+    component.ngOnInit();
+    fixture.whenStable().then(() => {
+      // 2 columns
+      expect(fixture.nativeElement.querySelectorAll('th').length).toBe(2);
+      // 1 head column and 2 data columns
```

Comment:
The comments seems off, I think you mean rows. It's the same for the other test cases.

Reply:
I think they are correct, as `th` is a `tableheader`.
Rows would be `tr`.

Reply:
Yes, that was my point, the comment is referring to https://github.com/eclipsesource/jsonforms/pull/1369/files/b4c01e46cabaeefad3b18fa40a208d5795078c3f#diff-bd6faf34e11b664459093ead7e1ebfecR164 isn't it?

Reply:
I see, I mixed up the loc, you are right. I will update it